generate-component-docs/generate.ts

```diff
@@ -80,10 +80,22 @@ export default () => {
 
   const checker = program.getTypeChecker();
 
+  function resolveDeclaration(exp: ts.Symbol) {
+    if (exp.valueDeclaration) {
+      return exp.valueDeclaration;
+    }
+
+    if (exp.declarations) {
+      return exp.declarations[0];
+    }
+
+    throw new Error('Could not resolve declaration');
+  }
+
```

Comment:
Declarations are marked as optional at a type level and there is [plenty of controversy around it](https://github.com/microsoft/TypeScript/issues/24706). Pulled out a separate function to resolve this consistently and is generating contracts as expected.